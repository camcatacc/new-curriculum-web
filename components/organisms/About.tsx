// Modules
import React from "react";

// Elements
import Paragraph from "components/atoms/Paragraph/Paragraph";
import AvatarWithName from "components/molecules/AvatarWithName";

// Definitions
export interface AboutProps {
	name: string;
	surname: string;
	paragraphs: string[];
}

// Element
const About = ({ name, surname, paragraphs }: AboutProps) => {
	return (
		<div style={{ gap: "10%" }} className="flex flex-col md:flex-row items-center">
			<div style={{ flex: 1 }} className="w-full md:w-auto">
				<AvatarWithName name={name} surname={surname} />
			</div>
			<div style={{ flex: 2 }}>
				{paragraphs.map((p) => (
					<div className="mb-8">
						<Paragraph>{p}</Paragraph>
					</div>
				))}
			</div>
		</div>
	);
};

export default About;
