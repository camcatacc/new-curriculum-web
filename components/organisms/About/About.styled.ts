import { motion } from "framer-motion";
import styled from "styled-components";

export const ParentContainer = styled.div`
	gap: 10%;
`;

export const FirstChildContainer = styled(motion.div)`
	flex: 1;
`;

export const SecondChildContainer = styled(motion.div)`
	flex: 2;
`;
