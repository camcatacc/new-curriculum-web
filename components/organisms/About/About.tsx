// Modules
import React from "react";
import { documentToReactComponents } from "@contentful/rich-text-react-renderer";
import { BLOCKS } from "@contentful/rich-text-types";

// Elements
import Paragraph from "components/atoms/Paragraph/Paragraph";
import AvatarWithName from "components/molecules/AvatarWithName/AvatarWithName";

// Definitions
import type { Document } from "@contentful/rich-text-types";

export interface AboutProps {
	name: string;
	surname: string;
	paragraphsDoc: Document;
}

// Element
const About = ({ name, surname, paragraphsDoc }: AboutProps) => {
	const options = {
		renderNode: {
			[BLOCKS.PARAGRAPH]: (_: any, children: any) => (
				<div className="mb-8">
					<Paragraph>{children}</Paragraph>
				</div>
			)
		}
	};
	const paragraphs = documentToReactComponents(paragraphsDoc, options);
	return (
		<div style={{ gap: "10%" }} className="flex flex-col md:flex-row items-center">
			<div style={{ flex: 1 }} className="w-full md:w-auto">
				<AvatarWithName name={name} surname={surname} />
			</div>
			<div style={{ flex: 2 }}>
				<div className="mb-8">{paragraphs}</div>
			</div>
		</div>
	);
};

export default About;
