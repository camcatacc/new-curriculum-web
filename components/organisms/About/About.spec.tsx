// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import About from "components/organisms/About/About";

// Definitions
import type { AboutProps } from "components/organisms/About/About";

// Default Props
const defaultProps: AboutProps = {
	name: "David",
	surname: "Camacho Cateura",
	paragraphsDoc: {
		data: {},
		content: [
			{
				data: {},
				content: [
					{ data: {}, marks: [], value: "I am a ", nodeType: "text" },
					{ data: {}, marks: [{ type: "bold" }], value: "commited", nodeType: "text" },
					{ data: {}, marks: [], value: ", capable and resolutive person. I have skills, experience and formation as a ", nodeType: "text" },
					{ data: {}, marks: [{ type: "bold" }], value: "developer", nodeType: "text" },
					{ data: {}, marks: [], value: ", which are very well complemented by my ", nodeType: "text" },
					{ data: {}, marks: [{ type: "bold" }], value: "business background", nodeType: "text" },
					{ data: {}, marks: [], value: ".", nodeType: "text" }
				],
				nodeType: "paragraph"
			},
			{
				data: {},
				content: [
					{
						data: {},
						marks: [],
						value: "I led an educational robotics and programming firm for five years. I devoted the last one and a half to ideate, design and develop an ",
						nodeType: "text"
					},
					{ data: {}, marks: [{ type: "bold" }], value: "online solution in the form of a hybrid app", nodeType: "text" },
					{ data: {}, marks: [], value: " and a web platform. ", nodeType: "text" }
				],
				nodeType: "paragraph"
			},
			{
				data: {},
				content: [
					{ data: {}, marks: [], value: "I am currently employed by Accenture as a ", nodeType: "text" },
					{ data: {}, marks: [{ type: "bold" }], value: "full-stack developer", nodeType: "text" },
					{ data: {}, marks: [], value: ".", nodeType: "text" }
				],
				nodeType: "paragraph"
			}
		],
		nodeType: "document"
	} as any
};

// Tests
describe("About", () => {
	it("Renders as expected", () => {
		const { container } = render(<About {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
});
