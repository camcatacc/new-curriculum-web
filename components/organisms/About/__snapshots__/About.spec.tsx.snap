// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Renders as expected 1`] = `
<div>
  <div
    class="sc-hKwDye hWqBml flex flex-col md:flex-row items-center"
  >
    <div
      class="sc-eCImPb jUvWMl w-3/5 md:w-auto"
    >
      <div
        class="sc-dkPtRN bjewRQ w-full relative mb-4"
      >
        <div
          style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
        >
          <noscript />
          <img
            alt="DavidCamacho Cateura"
            class="absolute top-0 left-0 bottom-0 right-0 rounded-full"
            data-nimg="true"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </div>
      <h3
        class="MuiTypography-root sc-gsDKAQ iRIomE MuiTypography-h3 MuiTypography-alignCenter"
      >
        David
         
        Camacho Cateura
      </h3>
    </div>
    <div
      class="sc-jRQBWg cWSwmB mt-4 md:mt-0 flex gap-8 flex-col justify-center"
    >
      <div
        class=""
      >
        <p
          class="MuiTypography-root sc-bdvvtL gedMyf MuiTypography-body1 MuiTypography-alignJustify"
        >
          I am a 
          <b
            style="color: rgb(255, 64, 129);"
          >
            commited
          </b>
          , capable and resolutive person. I have skills, experience and formation as a 
          <b
            style="color: rgb(255, 64, 129);"
          >
            developer
          </b>
          , which are very well complemented by my 
          <b
            style="color: rgb(255, 64, 129);"
          >
            business background
          </b>
          .
        </p>
      </div>
      <div
        class=""
      >
        <p
          class="MuiTypography-root sc-bdvvtL gedMyf MuiTypography-body1 MuiTypography-alignJustify"
        >
          I led an educational robotics and programming firm for five years. I devoted the last one and a half to ideate, design and develop an 
          <b
            style="color: rgb(255, 64, 129);"
          >
            online solution in the form of a hybrid app
          </b>
           and a web platform. 
        </p>
      </div>
      <div
        class=""
      >
        <p
          class="MuiTypography-root sc-bdvvtL gedMyf MuiTypography-body1 MuiTypography-alignJustify"
        >
          I am currently employed by Accenture as a 
          <b
            style="color: rgb(255, 64, 129);"
          >
            full-stack developer
          </b>
          .
        </p>
      </div>
    </div>
  </div>
</div>
`;
